src/Html/Parser.elm

```diff
@@ -267,6 +278,103 @@ element =
             )
 
 
+consumeJavaScriptUntilClosingTag : Parser (List Node)
+consumeJavaScriptUntilClosingTag =
+    Parser.loop [] <|
+        \acc ->
+            let
+                accumulate newNode =
+                    Parser.Loop <|
+                        case ( acc, newNode ) of
+                            ( [], first ) ->
+                                [ first ]
+
+                            ( (Text accChunk) :: tail, Text newChunk ) ->
+                                -- Merge top-most text node unless HTML comment nodes are interleaved
+                                Text (accChunk ++ newChunk) :: tail
+
+                            ( nonTextNode :: tail, _ ) ->
+                                newNode :: nonTextNode :: tail
+            in
+            Parser.oneOf
+                [ -- HTML comments are, albeit considered a bad practice recently,
+                  -- allowed inside <script> to hide scripts from really ancient web browser
+                  comment
+                    |> Parser.map accumulate
+                , Parser.lineComment "//"
+                    |> Parser.getChompedString
+                    |> Parser.map (Text >> accumulate)
+                , Parser.multiComment "/*" "*/" Parser.NotNestable
+                    |> Parser.getChompedString
+                    |> Parser.map (Text >> accumulate)
+                , javaScriptStringLike '"'
+                    |> Parser.map (Text >> accumulate)
+                , javaScriptStringLike '\''
+                    |> Parser.map (Text >> accumulate)
+                , javaScriptStringLike '`'
+                    |> Parser.map (Text >> accumulate)
+                , closingScriptTag
+                    |> Parser.map (\() -> Parser.Done (List.reverse acc))
+                , Parser.chompIf (always True)
+                    |> Parser.getChompedString
+                    |> Parser.map (Text >> accumulate)
+                ]
+
+
+closingScriptTag : Parser ()
+closingScriptTag =
+    Parser.token "</"
+        |. (Parser.chompWhile (\char -> char /= '>' && not (isSpaceCharacter char))
+                |> Parser.getChompedString
+                |> Parser.andThen
+                    (\chunk ->
+                        if String.toLower chunk == "script" then
+                            Parser.succeed ()
+
+                        else
+                            Parser.problem "not a </script>"
+                    )
+           )
+        |. Parser.chompWhile isSpaceCharacter
+        |. Parser.token ">"
+
+
+javaScriptStringLike : Char -> Parser String
+javaScriptStringLike terminatorChar =
```

Comment:
This is mostly based on https://github.com/elm/parser/blob/master/examples/DoubleQuoteString.elm